src/Search/SearchHandler.php

```diff
@@ -84,20 +85,31 @@ public function search(AdminInterface $admin, $term, $page = 0, $offset = 20)
         $datagridValues = $datagrid->getValues();
 
         $found = false;
+        $lastFilter = null;
         foreach ($datagrid->getFilters() as $filter) {
             /** @var FilterInterface $filter */
             $formName = $filter->getFormName();
 
+            // NEXT_MAJOR: Remove the `$filter instanceof ChainableFilterInterface` check.
+            if (null !== $lastFilter && $filter instanceof ChainableFilterInterface) {
```

Comment:
Shouldn't this code be inside the
```
if ($filter->getOption('global_search', false)) {
```
condition ?

By previousFilter, I assume you mean previous filter used for the search.